catalog/spatial/wps/spatial-wps-endpoint/src/main/java/org/codice/ddf/spatial/ogc/wps/process/endpoint/Validator.java

```diff
@@ -0,0 +1,260 @@
+                    inputDescription.getId());
+        }
+        BigInteger minOccurs = inputDescription.getMinOccurs() == null ?
+                BigInteger.ONE :
+                inputDescription.getMinOccurs();
+        if (inputs.size() < minOccurs.intValue()) {
+            throw new WpsException("Too few input items have been specified.",
+                    "TooFewInputs",
+                    inputDescription.getId());
+        }
+        inputs.forEach(input -> validateProcessInput(input, inputDescription));
+
+    }
+
+    public void validateProcessInput(Data inputs, DataDescription inputDescription) {
+        if (inputDescription.getDataFormats() != null && !inputDescription.getDataFormats()
+                .isEmpty()) {
+            validateDataFormats(inputDescription, inputs.getFormat());
+        }
+
+        if (inputDescription instanceof DataDescriptionGroup) {
+            if (inputs instanceof DataGroup) {
+                validateProcessInputs(((DataGroup) inputs).getData(),
+                        ((DataDescriptionGroup) inputDescription).getDataDescriptions());
+            } else {
+                throw new WpsException(
+                        "One or more of inputs for which the service was able to retrieve the data but could not read it.",
+                        "WrongInputData",
+                        inputDescription.getId());
+            }
+            //todo (RWY) - implement extended validation for LiteralDataDescription, ComplexDataDescription and BoundingBoxDataDescription when we understand the usecases better
+        } else if (inputDescription instanceof LiteralDataDescription) {
+
+        } else if (inputDescription instanceof ComplexDataDescription) {
+
+        } else if (inputDescription instanceof BoundingBoxDataDescription) {
+
+        }
+    }
+
+    /**
+     * @param dataDescription
+     * @param dataFormat
+     * @throws WpsException
+     */
+    public void validateDataFormats(DataDescription dataDescription, DataFormat dataFormat) {
+        if (dataDescription.getDataFormats() != null && !dataDescription.getDataFormats()
+                .isEmpty() //if there is no format specified don't worry about validating
+                && dataFormat != null
+                //if the given data format is null, don't validate we should just use the default.
+                && !dataDescription.getDataFormats()
+                .contains(dataFormat)) {
+            throw new WpsException(
+                    "One or more of the input or output formats specified in the request did not match with any of the formats defined for that particular input or output",
+                    "NoSuchFormat",
+                    dataDescription.getId());
+        }
+
+    }
+
+    /**
+     * @param outputDefinitions
+     * @param outputDescriptions
+     * @throws WpsException
+     */
+    public void validateProcessOutputs(Map<String, OutputDefinition> outputDefinitions,
```

Comment:
These parameter names are a bit confusing. Maybe `outputsRequest` and `outputsDefined/outputsProcess`?

Reply:
I changed outputDefinitions to requestedOutputs hopefully that clears up the confusion